Sources/SpeziOnboarding/OnboardingFlow/OnboardingStepIdentifier.swift

```diff
@@ -13,38 +13,103 @@ import SwiftUI
 /// An `OnboardingStepIdentifier` serves as an abstraction of a step in the onboarding flow as outlined within the ``OnboardingStack``.
 /// 
 /// It contains both the identifier for an onboarding step (the view's type) as well as a flag that indicates if it's a custom onboarding step.
-struct OnboardingStepIdentifier: Hashable, Codable {
-    let custom: Bool
-    let identifierHash: Int
-
+struct OnboardingStepIdentifier {
+    /// The source of the `OnboardingStepIdentifier`'s identity
+    enum IdentifierKind {
+        /// The `OnboardingStepIdentifier` derives its identity from a `View`'s type and source location
+        case viewTypeAndSourceLoc
+        /// The `OnboardingStepIdentifier` derives its identity from a `Hashable` value.
+        case identifiable(any Hashable)
+    }
+    
+    let identifierKind: IdentifierKind
+    
+//    let identifierHash: Int
+    
+    /// Whether the step is custom, i.e. created via e.g. ``OnboardingNavigationPath/append(customView:)``
+    let isCustom: Bool
```

Comment:
Might we be able to get rid of all `custom` flags? If the source loc is set, then it's not a custom view, if it's not set, it's a custom view?

Reply:
i was thinking about this as well but was kinda on the fence, but i think you're correct and we should do this.

Reply:
Cool!

Reply:
Sorry for throwing so many new requirements onto you, I just always hated the need for the "custom" flag and now we should have a fairly easy way of removing it.